x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/set.tsx

```diff
@@ -101,11 +95,114 @@ const fieldsConfig: FieldsConfig = {
   },
 };
 
+const getValueConfig: (
+  toggleCustom: () => void
+) => Record<
+  keyof ValueToggleFields,
+  {
+    path: string;
+    config?: FieldConfig<any>;
+    euiFieldProps?: Record<string, any>;
+    labelAppend: JSX.Element;
+  }
+> = (toggleCustom: () => void) => ({
+  value: {
+    path: 'fields.value',
+    euiFieldProps: {
+      'data-test-subj': 'valueFieldInput',
+    },
+    config: {
+      type: FIELD_TYPES.TEXT,
+      serializer: from.emptyStringToUndefined,
+      label: i18n.translate('xpack.ingestPipelines.pipelineEditor.setForm.valueFieldLabel', {
+        defaultMessage: 'Value',
+      }),
+      helpText: (
+        <FormattedMessage
+          id="xpack.ingestPipelines.pipelineEditor.setForm.valueFieldHelpText"
+          defaultMessage="Value for the field"
+        />
+      ),
+      fieldsToValidateOnChange: ['fields.value', 'fields.copy_from'],
+      validations: [
+        {
+          validator: ({ value, path, formData }) => {
+            if (isEmpty(value) && isEmpty(formData['fields.copy_from'])) {
+              return { path, message: 'A field value is required.' };
+            }
+          },
+        },
+      ],
+    },
+    labelAppend: (
+      <EuiButtonEmpty size="xs" onClick={toggleCustom} data-test-subj="toggleCustomField">
+        {i18n.translate('xpack.ingestPipelines.pipelineEditor.useCopyFromLabel', {
+          defaultMessage: 'Use copy_from field',
+        })}
+      </EuiButtonEmpty>
+    ),
+    key: 'value',
+  },
+  copy_from: {
+    path: 'fields.copy_from',
+    euiFieldProps: {
+      'data-test-subj': 'copyFromInput',
+    },
+    config: {
+      type: FIELD_TYPES.TEXT,
+      serializer: from.emptyStringToUndefined,
+      fieldsToValidateOnChange: ['fields.value', 'fields.copy_from'],
+      validations: [
+        {
+          validator: ({ value, path, formData }) => {
+            if (isEmpty(value) && isEmpty(formData['fields.value'])) {
+              return { path, message: 'A field value is required.' };
+            }
+          },
+        },
+      ],
+      label: i18n.translate('xpack.ingestPipelines.pipelineEditor.setForm.copyFromFieldLabel', {
+        defaultMessage: 'Copy from',
+      }),
+      helpText: (
+        <FormattedMessage
+          id="xpack.ingestPipelines.pipelineEditor.setForm.copyFromFieldHelpText"
+          defaultMessage="The origin field which will be copied to {field}."
```

Comment:
The current copy is okay, but I think we can shorten it.
```suggestion
          defaultMessage="Field to copy into {field}."
```